Tools/WinMLRunner/src/CommandLineArgs.cpp

```diff
@@ -255,7 +257,14 @@ CommandLineArgs::CommandLineArgs(const std::vector<std::wstring>& args)
             std::wcout << L"Version: " << pFileVersion << "." << pProductVersion << std::endl;
 
             delete[] pVersionData;
-            return;
+        }
+        else if ((_wcsicmp(args[i].c_str(), L"-InputImagePreprocess") == 0) && (i + 4 < args.size()))
+        {
+            m_inputImagePreprocess = true;
```

Comment:
CheckNextArgument(args, <index position>) can be used to check that the next 4 arguments aren't flags. Maybe extra checking will be needed to check if the next 4 arguments are numerical.

CheckNextArgument(args, i);
CheckNextArgument(args, i +1);
CheckNextArgument(args, i +2);
CheckNextArgument(args, i +3);

Reply:
Do you want me to add the check to see if it is numerical?  I can do that... but I don't think it would be very much more the wtof or sscanf, which would double the string interpretation...

Reply:
I would like some additional checking that the next 4 arguments are not flags.

As for the check to see if it's numerical, according to this: https://stackoverflow.com/questions/4654636/how-to-determine-if-a-string-is-a-number-with-c Seems like the most efficient way is to loop through the wstring until a non-digit character is found. [iswdigit](http://www.cplusplus.com/reference/cwctype/iswdigit/) could be used in a helper function to accomplish this. Thoughts?

Reply:
Already added CheckNextArugment checks to the code (in my local copy).  iwdigit doesn't handle negative or decimals... can search string for not of "0123456789.-", but that negative number matches flag for arugments... lol

Reply:
Does your feature's design suppoed to support negative numbers for the r/g/b factors?